src/libraries/System.Data.Common/src/System/Data/Filter/BinaryNode.cs

```diff
@@ -72,6 +76,8 @@ internal override bool DependsOn(DataColumn column)
             return _right.DependsOn(column);
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "Evaluating constant expression is safe.")]
```

Comment:
Would it be better if we marked these as RUC instead?

Maybe at a minimum, we should create a new method `EvalConstant();` that does:

1. Asserts it is a constant
2. Calls `Eval()`
3. Suppresses the warning from `Eval()`

And then call that from these places. That way we don't need to suppress IL2026 for these whole methods.